src/main/java/com/squareup/javawriter/builders/ParameterSpec.java

```diff
@@ -15,25 +15,59 @@
  */
 package com.squareup.javawriter.builders;
 
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
 import com.squareup.javawriter.TypeName;
 import com.squareup.javawriter.TypeNames;
+import java.lang.annotation.Annotation;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import javax.lang.model.element.Modifier;
 
 import static com.google.common.base.Preconditions.checkNotNull;
 
 /** A generated parameter declaration. */
 public final class ParameterSpec {
+  public final ImmutableList<AnnotationSpec> annotations;
+  public final ImmutableSet<Modifier> modifiers;
   public final TypeName type;
   public final Name name;
 
   private ParameterSpec(Builder builder) {
+    this.annotations = ImmutableList.copyOf(builder.annotations);
+    this.modifiers = ImmutableSet.copyOf(builder.modifiers);
     this.type = checkNotNull(builder.type);
     this.name = checkNotNull(builder.name);
   }
 
+  void emit(CodeWriter codeWriter) {
+    codeWriter.emitAnnotations(annotations);
+    codeWriter.emitModifiers(modifiers);
```

Comment:
I like the builder approach because `.build()` can throw if `modifiers` has inappropriate modifiers like `private`.